### PR TITLE
Integration bundle improvements

### PR DESCRIPTION
**Pull request by fedys**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      | 
| Related developer documentation PR URL | 
| Issue(s) addressed                     | 

#### Description:

Currently there is no validation for data type and/or length for fields when the "integration to Mautic" sync process runs. The lack of such validation may result in crashing of the synchronization. The user cannot see any of these errors. This PR tackles this issue by adding the data validation. In case there are any validation errors the given fields are not synced and a user is notified about the errors via UI notifications.

This PR also adds the following improvements:
1. Adds new integrations events: 
    - `'mautic.integration.INTEGRATION_BATCH_SYNC_COMPLETED_INTEGRATION_TO_MAUTIC'`
    - `'mautic.integration.INTEGRATION_BATCH_SYNC_COMPLETED_MAUTIC_TO_INTEGRATION'`
2. Multiselect custom fields DB type has been changed to `TEXT (65535)`

#### Steps to test this PR:

1. As there is no existing integration there is no way to test the changes in the integration bundle.
2. Make sure UI notification are working as expected.
3. Make sure Multiselect custom fields work as expected.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11303"></a>

### CODE REVIEW

**Comment by jos0405** [comment]
Hey @fedys ! What do I test here? :)

**Comment by fedys** [comment]
Hi @jos0405, I am about to fix the build issues and then I will mention the test steps (if any) in the description.

**Comment by escopecz** [comment]
@fedys I'm adding the WIP label until the PR description will be completed and CI passing so we know we don't have to check it until ready.

**Comment by fedys** [comment]
@jos0405 @escopecz The PR is ready to test.

**Review by escopecz — DISMISSED** [review]
I see no problems in the code. Most of the changes are in the IntegrationsBundle which the community cannot test as there is no public plugin using it. There are some changes for the internal notifications. Everything is well covered with tests and we use it in productions for months. Thanks Fedy! 👍 

@kuzmany could you test this with your integrations?

**Comment by RCheesley** [comment]
I think if @kuzmany can review/test this then we should be good to merge given that it has already been in production for some time, unless anyone else has plugins built on the IntegrationsBundle? @mzagmajster @biozshock maybe?

**Comment by mzagmajster** [comment]
Thanks for looking out for me @RCheesley. I do not think any of my projects will be effected by this change.

**Comment by RCheesley** [comment]
Ah thanks @mzagmajster - we just need folks who are using the IntegrationsBundle in plugins to test some PRs, as currently we don't have any public plugins to test with. Wasn't sure if any of yours use it yet.

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php [hunk 1]
```diff
@@ -381,4 +381,18 @@ public function getFieldsByType($type)
...
         return $this->findBy(['type' => $type]);
     }
+
+    /**
+     * @return mixed[]
+     */
+    public function getFieldSchemaData(string $object): array
+    {
+        return $this->_em->createQueryBuilder()
+            ->select('f.alias, f.label, f.type, f.isUniqueIdentifer, f.charLengthLimit')
```
Comment (kuzmany) [inline app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: Looks like charLengthLimit  columns is from another PR what is not pushed to community yet.

**Review by kuzmany — CHANGES_REQUESTED** [review]

**Comment by kuzmany** [comment]
I have used it in our project because we need process objects after sync.

`IntegrationEvents::INTEGRATION_BATCH_SYNC_COMPLETED_INTEGRATION_TO_MAUTIC` and             `IntegrationEvents::INTEGRATION_BATCH_SYNC_COMPLETED_MAUTIC_TO_INTEGRATION`  works pretty well, then If we resolve conflicts and resolve reported issue above we can merge it

File: app/bundles/LeadBundle/Entity/LeadFieldRepository.php (hunk 1 above)
Comment (fedys) [reply app/bundles/LeadBundle/Entity/LeadFieldRepository.php]: Good catch! Thanks @kuzmany!

**Comment by fedys** [comment]
@kuzmany I resolved the issue and the conflicts.

**Review by escopecz — APPROVED** [review]

**Pull request merged**